### DISCUSSION TITLE
Difference between patch_size and val_patch_size

### DISCUSSION THREAD

**Discussion by AceMcAwesome77** [Project-MONAI/tutorials#1468] [General]
Hi, I am implementing the RetinaNet3D tutorial and I'm not sure why patch_size and val_patch_size are different in the detection/config/config.json file (patch size is 192,192,80 and val_patch_size is 512,512,208).  Wouldn't we want to be using the same patch size for training as we do for validation and testing?  I would also think that with a certain matrix size going into the neural network (patch_size), the input from any data being inferenced by the trained model would need to be the same size to fit into the input layer.  Why is that not the case?  Thanks!

**Comment by KumoLiu** [comment]
Hi @Can-Zhao, could you share some comments on this question?
Thanks in advance!

**Comment by Can-Zhao** [comment]
The network is a CNN. The patch size of training and inference do not have to be the same. To my experience, during inference, as long as the patch size is no smaller than training patch size, it would work well.

**Reply by AceMcAwesome77** [reply]
Thanks for the reply!  That is a counterintuitive to me but I do see other forums saying that the network will work with different training and inference image size as long as there are no Dense layers.  Am I right in thinking that the optimal setting is still to have training and inference patch sizes be the same, even at a cost of longer inference time?  Or does it really not matter at all to the results?

**Reply by Can-Zhao** [reply]
No worries. It does not matter as long as inference image size is larger. The opposite won't work.

**Reply by AceMcAwesome77** [reply]
Does it matter if the ratio of the X and Y patch_size dimensions does not remain the same?  For example, if the training patch_size is (128, 128, 160) but the inference patch_size is (310, 340, 320) for coordinates X, Y, Z.  Will this cause any degradation in performance?

**Reply by Can-Zhao** [reply]
No, it does not matter. Inference with large patch is essential the same as sliding window inference using small patches.

**Reply by AceMcAwesome77** [reply]
Is this also true of the training patch_size?  As in, is there any difference in model training accuracy when using a larger or smaller train patch_size?  Particularly for large objects that may not fit within an entire train patch window of (192,192,80).

**Reply by Can-Zhao** [reply]
Yes, there is difference in model training accuracy when using a larger or smaller train patch_size. Larger train patch_size gives better accuracy if properly trained. But smaller train patch_size is easier to train.
